### DISCUSSION TITLE
Using https for HMR with puma-dev

### DISCUSSION THREAD

**Discussion by ledermann** [ElMassimo/vite_ruby#215] [Show and tell]
Hey @ElMassimo, thanks for this great project! I'm trying to migrate some Rails projects from `webpacker` to `vite_ruby` and it seems a bit hard to configure Vite for `https` with `puma-dev` in development.

In the `puma-dev` docs there is a recipe for `webpack-dev-server` (via proxy):
https://github.com/puma/puma-dev#webpack-dev-server

I was successful to adopt this for Vite with the following configuration, which I would like to share:

Set up `puma-dev`:
```bash
puma-dev link -n myapp
echo 3036 > ~/.puma-dev/vite.myapp
```

Edit **vite.json**:
```json
  "development": {
    "autoBuild": true,
    "publicOutputDir": "vite-dev",
    "host": "vite.myapp.test",
    "port": 3036
  },
```

Add this **vite.config.ts**:
```ts
server: {
  hmr: {
    host: 'vite.myapp.test',
    clientPort: 443,
  },
},
```

Edit `content_security_policy.rb`:
```ruby
Rails.application.configure do
  config.content_security_policy do |policy|
    if Rails.env.development?
      policy.style_src :self, :unsafe_inline
      policy.script_src :self, :unsafe_eval, :unsafe_inline, :blob, "https://#{ViteRuby.config.host}"
      policy.connect_src :self, "wss://#{ViteRuby.config.host}"
      # ...
```

Now the app is available locally at `https://myapp.test` using HMR via `https`, too.

**Comment by ElMassimo** [comment]
Hi Georg! Glad that you find it useful, and thank you for [the Rails version of `pingcrm`](https://github.com/ledermann/pingcrm) 😃 

Thanks for sharing this recipe! I've added [a link](https://vite-ruby.netlify.app/guide/troubleshooting.html#making-hmr-work-with-puma-dev-in-https-apps) in the [_FAQs_](https://vite-ruby.netlify.app/guide/troubleshooting.html) to this discussion.

**Reply by ledermann** [reply]
Great, thanks!

**Reply by jisuanjixue** [reply]
Thanks for  @ledermann I use your [the Rails version of templatus-inertia](https://github.com/ledermann/templatus-inertia) add puma-dev , There are some errors, I don't know where

<img width="1316" alt="屏幕快照 2022-11-23 下午10 41 51 上午" src="https://user-images.githubusercontent.com/24429825/203464127-0fdf1d15-956b-4824-ab8f-397bc67700d9.png">
<img width="1839" alt="屏幕快照 2022-11-23 下午10 41 40 上午" src="https://user-images.githubusercontent.com/24429825/203464144-1215dd6b-9cd2-462f-9e6f-d184f0c917f2.png">

 I went wrong, I don't know much about some back-end technologies, I hope you can help me take a look.

**Reply by ledermann** [reply]
@jisuanjixue Seems there is something wrong with your `puma-dev` setup, see the [Getting Started](https://github.com/templatus/templatus-inertia#getting-started) section in my repo. Please open a new issue in [my repo](https://github.com/templatus/templatus-inertia) if it still fails. 

Here we are in the **vite_ruby** repo, which appears to be unrelated.

**Reply by jisuanjixue** [reply]
@ledermann  OK,  thanks!

**Reply by whatcould** [reply]
This is very helpful! I want to add that for some reason I had `"skipProxy": true,` in my vite.json, which seems to have meant that vite-rails was specifying the host and port, eg `https://vite.myapp.test:3036/...`. Including the port number threw off the SSL and also the content-security policy. After an hour I removed `skipProxy`, restarted Vite, and it worked.
